### ISSUE TITLE
Remove version checks in `brian2.__init__.py`?

### ISSUE THREAD

**Issue by mstimberg** [brian-team/brian2#1443]
When importing the `brian2` package, we check a few packages for outdated versions:
https://github.com/brian-team/brian2/blob/4886c46e92d9197cbc7401fd2ab6debe9116134d/brian2/__init__.py#L74-L100

I'm not entirely convinced anymore that this is a good idea, so maybe we should simply remove these checks. But it would be good to have a look first how/if other packages are handling this. If we keep this, we should at least make sure that the checked requirements stay in sync with the actual requirements, by reading them in from some canonical source (`requirements.txt`, `pyproject.toml`, ...).

**Comment by bdevans** [comment]
Yes, I agree that this needs to be read in from another source if you want to keep it but provided the installed packages meet the minimum requirements to install & run Brian, maybe this should just be left to `pip` or `conda` to nag users to upgrade? The main reason I can imagine for this scenario (recommending but not requiring an upgrade) would be for better performance, however if it makes that much of a difference, maybe we should just bump the minimum requirements?

**Comment by maverick4code** [comment]
@mstimberg  Yes, even I think the optimal thing will be to remove the version checks from `brian2.__init__.py`, relying on `pyproject.toml` and package managers for dependency management.

**Comment by Legend101Zz** [comment]
Hi @mstimberg , I noticed this issue has been open for quite a while now. Is this something you'd still like to see addressed despite being around for 3 years? If yes, I'd love to work on it and have a few ideas to propose:

1. We could completely remove the version checks since pip/conda already handle dependencies well these days
2. We could keep the checks but make them read from your canonical requirements source (pyproject.toml/requirements.txt)
3. Or update the current checks to reflect the latest minimum versions needed

 Let me know if you'd like me to prepare a PR with one of these approaches!

**Comment by mstimberg** [comment]
@Legend101Zz. Hmm, yes, I think we should probably simply remove these checks. I think the main use case was for things lke developers simply opening `brian2` in an IDE, which would add it to the path when running things, instead of doing a "proper" `pip install -e .` or something along these lines. These days, you could use our devcontainer, or an IDE would probably display some warning about outdated dependencies after reading them in from `pyproject.toml`. Still, I know that some projects have these runtime checks for some very important dependencies, e.g. `scipy` does it for `numpy`:
https://github.com/scipy/scipy/blob/41641c2574363ccde8a9177004082e12e18aef71/scipy/__init__.py#L68-L74
Not entirely use what their reasoning behind this is.

**Comment by Legend101Zz** [comment]
Thanks for the response, @mstimberg ! That makes sense. I’ll go ahead and submit a patch in the next 1–2 days :)

**Comment by Legend101Zz** [comment]
Also, @mstimberg , are there any larger issues or features you’d recommend I take on next? I’d love to contribute more meaningfully and get a deeper understanding of Brian in the process.

**Comment by mstimberg** [comment]
Hi @Legend101Zz. Maybe you could have a look at #1368? We haven't heard back in a while from the person that expressed interest on working on it, so I think it is free for taking. 

There are also a few "boring" infrastructure tasks that need someone helping out. For example, at some point we need to  properly deal with the fact that we are using internal functions from `setuptools` (which keep moving around), see #1558.

**Comment by Legend101Zz** [comment]
> Hi [@Legend101Zz](https://github.com/Legend101Zz). Maybe you could have a look at [#1368](https://github.com/brian-team/brian2/issues/1368)? We haven't heard back in a while from the person that expressed interest on working on it, so I think it is free for taking.
> 
> There are also a few "boring" infrastructure tasks that need someone helping out. For example, at some point we need to properly deal with the fact that we are using internal functions from `setuptools` (which keep moving around), see [#1558](https://github.com/brian-team/brian2/pull/1558).

Thanks a lot for pointing these out! And apologies for the delayed response — I was caught up with some research work over the past few days.

I'll dive into the issue and aim to push a patch soon. If you have a bit of bandwidth to review it when it's ready, that’d be great — we can get it wrapped up and move on to the next one. Looking forward to contributing more! Cheers!
